src/test/java/htsjdk/samtools/cram/compression/rans/RansTest.java

```diff
@@ -160,8 +206,31 @@ private static void roundTripForOrder(final ByteBuffer data, final RANS.ORDER or
         Assert.assertFalse(uncompressed.hasRemaining());
     }
 
-    private static void roundTripForOrder(final byte[] data, final RANS.ORDER order) {
-        roundTripForOrder(ByteBuffer.wrap(data), order);
+    public ByteBuffer ransBufferMeetBoundaryExpectations(
+            final int size,
+            final RANSEncode ransEncode,
+            final RANSDecode ransDecode,
+            final RANSParams params){
+        // helper method for Boundary Expectations test
+        final ByteBuffer raw = ByteBuffer.wrap(randomBytesFromGeometricDistribution(size, 0.01));
+        final ByteBuffer compressed = ransEncode.compress(raw, params);
+        final ByteBuffer uncompressed = ransDecode.uncompress(compressed,params);
+        Assert.assertFalse(compressed.hasRemaining());
```

Comment:
The asserts should be immediately after the code they're checking, i.e., presumably this is testing that the compress call consumed all the data - if so, it should be moved to right after the compress call. No sense in uncompressing if the compress criteria weren't met.